cmake/everest-generate.cmake

```diff
@@ -494,7 +499,82 @@ function (ev_add_py_module MODULE_NAME)
             add_subdirectory(${MODULE_PATH})
         endif()
     else()
-        message(WARNING "C++ module ${MODULE_NAME} does not exist at ${MODULE_PATH}")
+        message(WARNING "Python module ${MODULE_NAME} does not exist at ${MODULE_PATH}")
+        return()
+    endif()
+
+    # FIXME (aw): this will override EVEREST_MODULES, might not what we want
+    set_property(
+        GLOBAL
+        PROPERTY EVEREST_MODULES ${EVEREST_MODULES}
+    )
+endfunction()
+
+function (ev_add_rs_module MODULE_NAME)
+    set(EVEREST_MODULE_INSTALL_PREFIX "${CMAKE_INSTALL_LIBEXECDIR}/everest/modules")
+
+    set(MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/${MODULE_NAME}")
+    set(MODULE_BUILD_PATH "${CMAKE_CURRENT_BINARY_DIR}/${MODULE_NAME}")
+
+    if(IS_DIRECTORY ${MODULE_PATH})
+        if(NOT ${EVEREST_ENABLE_RS_SUPPORT})
+            message(STATUS "Excluding Rust module ${MODULE_NAME} because EVEREST_ENABLE_RS_SUPPORT=${EVEREST_ENABLE_RS_SUPPORT}")
+            return()
+        elseif("${MODULE_NAME}" IN_LIST EVEREST_EXCLUDE_MODULES)
+            message(STATUS "Excluding module ${MODULE_NAME}")
+            return()
+        elseif(EVEREST_INCLUDE_MODULES AND NOT ("${MODULE_NAME}" IN_LIST EVEREST_INCLUDE_MODULES))
+            message(STATUS "Excluding module ${MODULE_NAME}")
+            return()
+        else()
```

Comment:
I'm not really happy with this code duplication for all different `ev_add_*_module` function

Reply:
I added a todo for now.